src/components/GraphNode/_GraphNodeStatic.js

```diff
@@ -0,0 +1,241 @@
+  display: inline-block;
+  text-align: center;
+  padding: 2px 10px;
+  max-width: 100%;
+  pointer-events: all;
+
+  ${props =>
+    !props.highlighted &&
+    `
+      white-space: nowrap;
+    `};
+`;
+
+const LabelStandard = LabelTemplate.extend`
+  color: ${props =>
+    props.contrastMode
+      ? props.theme.colors.black
+      : props.theme.colors.purple800};
+  font-size: ${props => props.theme.fontSizes.normal};
+  margin-top: 4px;
+`;
+
+const LabelMinorStandard = LabelTemplate.extend`
+  color: ${props =>
+    props.contrastMode
+      ? props.theme.colors.black
+      : props.theme.colors.purple600};
+  font-size: ${props => props.theme.fontSizes.small};
+  margin-top: -7px;
+`;
+
+class GraphNodeStatic extends React.PureComponent {
+  findFirstSearchMatch = (searchTerms, text) => {
+    let match = null;
+    searchTerms.forEach(term => {
+      const start = text.search(term);
+      if (isEmpty(match) && start !== -1) {
+        match = { start, length: term.length };
+      }
+    });
+    return match;
+  };
+
+  renderSvgLabels() {
+    return (
+      <SvgTextContainer y={this.props.labelOffset}>
+        <LabelSvg contrastMode={this.props.contrastMode}>
+          {this.props.label}
+        </LabelSvg>
+        <LabelMinorSvg contrastMode={this.props.contrastMode}>
+          {this.props.labelMinor}
+        </LabelMinorSvg>
+      </SvgTextContainer>
+    );
+  }
+
+  renderStandardLabels() {
+    const {
+      label,
+      labelMinor,
+      highlighted,
+      contrastMode,
+      labelOffset,
+      searchTerms,
+    } = this.props;
+
+    return (
+      <foreignObject
+        y={nodeBaseSize / 2 + labelOffset}
+        x={-0.5 * labelWidth}
+        width={labelWidth}
+        height="200px"
+        style={{
+          textAlign: 'center',
+          pointerEvents: 'none',
+        }}
+      >
+        <LabelsStandardContainer>
+          <LabelStandard highlighted={highlighted} contrastMode={contrastMode}>
+            <MatchedText
+              noBorder
+              text={label}
+              match={this.findFirstSearchMatch(searchTerms, label)}
+            />
+          </LabelStandard>
+          {!isEmpty(labelMinor) && (
+            <LabelMinorStandard
+              highlighted={highlighted}
+              contrastMode={contrastMode}
+            >
+              <MatchedText
+                noBorder
+                text={labelMinor}
+                match={this.findFirstSearchMatch(searchTerms, labelMinor)}
+              />
+            </LabelMinorStandard>
+          )}
+
+          {this.props.renderAppendedInfo()}
+        </LabelsStandardContainer>
+      </foreignObject>
+    );
+  }
+
+  render() {
+    const { id } = this.props;
+    const Shape = shapes[this.props.shape];
+
+    return (
+      <GraphNodeWrapper
+        onMouseEnter={ev => this.props.onMouseEnter(id, ev)}
+        onMouseLeave={ev => this.props.onMouseLeave(id, ev)}
+        onClick={ev => this.props.onClick(id, ev)}
```

Comment:
I don't know exactly what performance symptoms are being seen but if there is a large number of nodes, I have had problems in the past with garbage collection happening during animations and causing lag. 

One way to lower the amount of garbage collection that occurs is to reduce the amount of inline functions in render as a new function object is recreated every render, more objects === more garbage collection. 

Might not be helpful in this instance but something to think about if there are still janks in animations.